spring-jdbc/src/main/java/org/springframework/jdbc/support/incrementer/H2SequenceMaxValueIncrementer.java

```diff
@@ -47,7 +47,7 @@ public H2SequenceMaxValueIncrementer(DataSource dataSource, String incrementerNa
 
 	@Override
 	protected String getSequenceQuery() {
-		return "select " + getIncrementerName() + ".nextval from dual";
+		return "select next value for " + getIncrementerName() + " from dual";
```

Comment:
Please, use standard-compliant `VALUES NEXT VALUE FOR sequenceName`. All other constructions aren't guaranteed to work in all compatibility modes of all versions of H2.

`SELECT … FROM DUAL` is also an Oracle compatibility feature that may be rejected by H2 or interpreted in some unexpected way.

Reply:
@katzyn Thanks! Changed.